packages/dds/tree/src/simple-tree/objectNode.ts

```diff
@@ -112,12 +119,21 @@ export type FieldHasDefault<T extends ImplicitFieldSchema> = T extends FieldSche
  * To prevent extraneous properties in literals for the fields of an empty object from compiling, the empty case is special cased to produce `Record<string, never>`.
  * More details at {@link https://mercury.com/blog/creating-an-emptyobject-type-in-typescript}.
  *
+ * Additionally when T is exactly `RestrictiveStringRecord<ImplicitFieldSchema>` produce just `never` so that it is assignable to the insertable for any given object type.
+ *
+ * Separating `{}` from `RestrictiveStringRecord<ImplicitFieldSchema>` is a bit messy since both extend each-other despite them being very different types.
+ * A third dummy type `{ x: "x" }` is used to resolve this.
+ *
  * @system @public
  */
 export type InsertableObjectFromSchemaRecord<
 	T extends RestrictiveStringRecord<ImplicitFieldSchema>,
-> = Record<string, never> extends T
-	? Record<string, never>
+> = RestrictiveStringRecord<ImplicitFieldSchema> extends T
+	? { x: "x" } extends T
```

Comment:
Nit: It would probably be worth being a bit more explicit here that this is a dummy value, since `x: "x"` ends up in API reports and such. Maybe `_dummy "DUMMY"` or something?

Reply:
replaced with { arbitraryKey: "arbitraryValue" }